### ISSUE TITLE
Design new component for double coil toggling

### ISSUE THREAD

**Issue by zepumph** [phetsims/faradays-law#99]
From many design meetings, the most recent of which was this afternoon, we want mock ups for a new control for the coils. We want two ideas:

1. A checkbox like the bottom left of Function Builder, this checkbox is "button-like" and has an icon instead of text.

2. An a/b switch type thang that looks like the bottom of expression exchange.

Here is a mockup of the balloon toggle in BASE that @terracoda made. 

https://docs.google.com/document/d/1yDFeJYieDX-_EdylVtHymFtFbr5CoxNSJyobxOirNEQ/edit

**Comment by arouinfar** [comment]
Here are some options to discuss in today's meeting. 

(1) Checked scene somewhat similar to @terracoda's mockups
![image](https://user-images.githubusercontent.com/8419308/34169000-32224fae-e4a3-11e7-8ed8-87bcc692fbd5.png)
![image](https://user-images.githubusercontent.com/8419308/34169012-3d253380-e4a3-11e7-8676-7a4c75b14920.png)

(2) Button with changing appearance (like Reveal button or play/pause)
![image](https://user-images.githubusercontent.com/8419308/34169033-4e090294-e4a3-11e7-991a-eff412b9857c.png)
![image](https://user-images.githubusercontent.com/8419308/34169044-57c60fca-e4a3-11e7-8109-b90f0ffc469d.png)


(3) Visual checkbox like Function Builder
![image](https://user-images.githubusercontent.com/8419308/34169073-6ce6c57a-e4a3-11e7-8e82-d394ea250295.png)



(4) Toggle where coils same size/alignment as original scenes
![image](https://user-images.githubusercontent.com/8419308/34169095-78f28502-e4a3-11e7-886a-642ab9325e21.png)


(5) Toggle with different sizing/alignment
![image](https://user-images.githubusercontent.com/8419308/34169102-81c17d6e-e4a3-11e7-8be4-88d0db6a8a48.png)

**Comment by arouinfar** [comment]
Given the number of absences, discussion has been deferred to the next keyboard navigation meeting (Jan 2).

I think options (3) and (5) clearly convey what the control does, are prominent enough for the desired flow, and are more aesthetically pleasing than the other options. We will need to have a discussion to decide which control (checkbox vs. ab switch) is better suited the the goals and interaction.

Tagging interested parties @amanda-phet @ariel-phet @emily-phet @terracoda @zepumph @jessegreenberg @oliver-phet.

**Comment by amanda-phet** [comment]
Unfortunately I missed a portion of the conversation last week, so I'm not sure exactly what was discussed or decisions that were made. When I left, it sounded like we would try to create a custom switch component so that we can continue to use switches, or dual scene selection style buttons, without having to limit ourselves to them behaving like checkboxes or radio buttons under the hood. 

So, it seems like we still need to decide which route we are going before we design it, right?

**Comment by arouinfar** [comment]
@amanda-phet, as far as I understand, no decisions were made other than to mock up the various options that have been tossed around. The purpose of the mockups is to provide a visual aid to facilitate these conversations, with the hope that they may help us reach some sort of consensus.

**Comment by terracoda** [comment]
In case it is helpful, here's what Amy's mock-ups might sound like with a screen reader:
(1, 3, 4, and 5) As checkboxes describing experiments
 - Two-coil experiments, checkbox, not checked
 - On-demand help text: Play with two coils or just one. 
   - Interaction: Two-coil experiments, checked
     - Alert: Small coil with two rings added to Play Area.
   - Interaction: Two-coil experiments, not checked
     - Alert: Small coil with two rings removed from Play Area.
   
(2) As two buttons describing adding and removing a small coil
 - Add small coil, button
 - Remove small coil, button
   - On-demand help text: Play with two coils or just one. 
   - Alert: Small coil with two rings added to Play Area.
   - Alert: Small coil with two rings removed from Play Area.
 
**Note 1:** Alerts could be the same for all.

**Note 2:** The ARIA role, `role="switch",` is not yet supported by all browsers and AT, but with the switch role, a checkbox could/might be made to sound like this:**
 - Two-coil experiments, switch, off
 - On-demand help text: Play with two coils or just one. 
   - Interaction: Two-coil experiments, on
     - Alert: Small coil with two rings added to Play Area.
   - Interaction: Two-coil experiments, off
     - Alert: Small coil with two rings removed from Play Area.

**Comment by emily-phet** [comment]
I can answer any questions about this tomorrow - but essentially, a checkbox is not an option for Faraday's Law coil control. Kathy pointed out in the last Thursday design meeting on this (Dec. 14th, I believe), a relevant and ongoing issue with checkboxes - that users may select them (making them "checked") but they don't "uncheck" them as readily. That did not seem consistent with the desired flow for this sim. So, while a checkbox in general may functionally achieve the desired task, it's use by visual users is not consistent with the desired goals of use within the context of this sim. 

The desired behavior, from that discussion, seems more consistent with wanting to present two different options for users to alternate between at will (consistent with use of an A/B switch)...which is not the same as having one option on or off (the checkbox solution). So we will need an A/B switch here.

**Comment by arouinfar** [comment]
Thanks for your comments @emily-phet. We certainly should discuss a bit more at the next design/keyboard meeting. However, I want to push back a little on this:
> Kathy pointed out in the last Thursday design meeting on this (Dec. 14th, I believe), a relevant and ongoing issue with checkboxes - that users may select them (making them "checked") but they don't "uncheck" them as readily. That did not seem consistent with the desired flow for this sim.

While I agree that checkboxes aren't as readily "unchecked", I don't think it's as much of an issue in this particular sim (though we should certainly be cognizant of this moving forward). If students discover the two-coil experiment checkbox a little later in their exploration, it's not pedagogically problematic to leave it checked. As @ariel-phet pointed out during the previous design meeting, the two-coil experiment is a bit unique, as it doesn't completely replace the one-coil experiment -- students still have access to the original coil. I think the real issue is to make sure that a two-coil checkbox wouldn't be attractive enough to throw off the desired flow (interacting with magnet first).

**Comment by terracoda** [comment]
@emily-phet and @arouinfar these are excellent points. In my very interviews for BASE, blind users did active the two-balloon experiment button. There were complicating issue regarding the balloon interaction which no longer exist, but something to keep on eye on for sure.

**Comment by emily-phet** [comment]
Thanks @arouinfar and @terracoda for further input. I think we should just clarify with @ariel-phet and @kathy-phet what the ideal use case would be for this sim. I think that would help us understand what the best fit is for a visual cue, which will help with all of the other design decisions. 

Just to make a note of it somewhere, here is my assumed intended use of the three "toggle" options in Faraday's Law:
- Field Lines checkbox: Intended to be one of the last things students explore. If it's one of the last things explored, leaving on is not problematic (e.g., wouldn't be visually overwhelming at that point). Visual checkbox appropriate since users tend to select options at edge of screen later in exploration, and ok if they leave it on if using later in sim use.
- Flip magnet button: probably ok to be explored at any point in sim use, needs to have the affordance of being able to rapidly flip back and forth (precludes toggle visual which may cue users they need to individually select between left side/right side of a toggle switch to quickly flip back and forth). Button ideal for this rapid flip back/forth option since single "activate" can be done repeatedly without changing cursor/finger/keypress location.
- One Coil/Two Coil options: (EM's presumed goal) is to support cueing to go back and forth between these options, or at least, to discourage bringing up the second coil and leaving it there for an extended time. Not sure if it's important that second coil be added late in exploration, or if it's all the same if it happens early or late. (need further clarification from AP and KP on desired flow).

**Comment by ariel-phet** [comment]
A few points on the above discussion.

As to @emily-phet question about flow -- Agreed on the Field Lines checkbox and Flip Magnet button. However, for the one/two coil option, I don't think it is necessary to go back and forth between the two options. I suppose the idealized flow for the sim in regard to the coils would be something like, _oh look when I play with the single coil, what is affecting the bulb brightness? It seems like the speed at which I move the magnet appears to be a determining factor and maybe if I am close or far away from that coil. Now when i turn on the second coil, there seems to be a difference in the reaction from each coil to a similar magnet motion. What is different between the coils? Oh one has more loops than the other_ So I think that once the second coil is "in play" the sim can be left in that state for comparison between the two coils. It may be valuable to switch back and see that basically with just the first coil, or with both coils, the same action on the first coil produces essentially the same result. but flipping back and forth is not hugely important.

As for the mockups from @arouinfar - from a visual standpoint, I think if we go with some kind of a/b switch, it will not work well to be "floating" in this case. It will need to be contained or such in a box or a background so it does not look like it is in the play area. When the images for the A/B switch are reasonably large (as I think they need to be) those start to feel like mini coils in the play area. I doubt anyone would actually think they were mini coils, but it seems to violate the feel of the play area.

**Comment by emily-phet** [comment]
@ariel-phet thanks for the clarification! In that case, what about a floating panel in the lower left of the screen that contains 1) a checkbox with an image of a two turn coil and below that 2) a checkbox with the text "field lines". The coil in this floating panel could be small. Then the flip magnet button could be made a touch smaller (it's pretty huge...I think to match the visual weight of the current two coil button).

**Comment by arouinfar** [comment]
Here are some mockups of @ariel-phet and @emily-phet's suggestions. I think it would be nice to bring this back to keyboard nav meeting once the static description work settles down.

![image](https://user-images.githubusercontent.com/8419308/38338596-69cf2ef4-3827-11e8-868a-25ad6a60ed03.png)

I think a string-less checkbox looked a bit odd in the panel, so I mocked up a version with some text, as well.
![image](https://user-images.githubusercontent.com/8419308/38338605-71bf8168-3827-11e8-94db-bc5fac94c2eb.png)

**Comment by jhung** [comment]
I have come up with some different designs for a toggle. In these designs I chose to follow the existing design of showing 1 and 2 coils as the possible states.

Any of these look promising?

![phet-faraday-coil-toggle-v01-jh](https://user-images.githubusercontent.com/564762/42779508-aaf92712-890e-11e8-8563-60d06fb9959a.png)

**Comment by emily-phet** [comment]
It was decided to maintain switch style currently - may revisit when full description design is revisited.
